### PR TITLE
her and her2 ILP64

### PR DESCRIPTION
**Pull request by NaveenElumalaiAMD**
resolves: [LWPMLSE-594](https://ontrack-internal.amd.com/browse/LWPMLSE-594)

### CODE REVIEW

File: clients/gtest/blas2/her2_gtest.yaml
Comment (TorreZuk) [inline clients/gtest/blas2/her2_gtest.yaml]: Talk to Daine and propose a constants in common for this 2147483649 and 666666.  I think we lean toward just large enough to trigger loops rather than int32 overflow unless your tests are fast.   Then we can get just edit common to really test large on the rare occasion but for now leave actual stress tests to rocBLAS by default.   These are problematic enough I think

File: clients/gtest/blas2/her_gtest.yaml
Comment (TorreZuk) [inline clients/gtest/blas2/her_gtest.yaml]: I would tend to smaller like -0.5 and 0.5 as we expect larger sums but these are fine if all passing

File: clients/gtest/blas2/her_gtest.yaml
Comment (TorreZuk) [inline clients/gtest/blas2/her_gtest.yaml:34]: We can argue fortran _64 suffices as it invokes C function only would miss double negation bug which would be very low probability.

**Review by TorreZuk — APPROVED** [review]
All looks good to me but get Daine's approval before merge, the V2 I am unsure of.

File: library/src/nvidia_detail/hipblas.cpp
Comment (daineAMD) [inline library/src/nvidia_detail/hipblas.cpp:10341]: should be int64_t, along with others further down.

**Review by daineAMD — APPROVED** [review]
Looks good aside from comment and testing discussion we had today.

File: library/include/hipblas.h
Comment (daineAMD) [inline library/include/hipblas.h:8661]: Actually this was changed to be *_v2_64 for other functions so should be changed here too.

**Comment by TorreZuk** [comment]
Windows failure can be ignored but it interesting to see it fail later than first tests with that stack.

**Pull request merged**
